provider/blockdev_base.py

```diff
@@ -95,7 +95,11 @@ def generate_data_file(self, tag, filename=None):
         timeout = params.get_numeric("create_tempfile_timeout", 720)
         backup_utils.generate_tempfile(
             self.main_vm, self.disks_info[tag][1], filename, image_size, timeout)
-        self.files_info[tag] = [filename]
+
+        if tag not in self.files_info:
```

Comment:
Every test, we will create tmp files on base and snapshot, so there's no scenario that we create two tmp files on on tag. So I think self.files_info[tag] = [filename] is just enough.